### ISSUE TITLE
Suggestion: InfoBar Severity Colours - problems with Hue in Light and Dark Themes

### ISSUE THREAD

**Issue by mdtauk** [microsoft/microsoft-ui-xaml#3744]
# Suggestion: InfoBar Severity Colours - problems with Hue in Light and Dark Themes

**Describe the bug**

When looking at the colours used for the InfoBar Backgrounds in the Dark Theme, compared to the Foreground, and both Background and Foreground colours in Light Theme - The hues seem to shift radically to the point where the meaning for the colours, and the ability to discern Success, Critical and Warning severity is hampered.

**Expected behavior**

Green for Success **not Olive**
Yellow for Warning **not Orange**
Red for Critical **this colour fits well**

**Screenshots**

![Current Background and Foreground severity colours](https://user-images.githubusercontent.com/7389110/100917791-8b027200-34cf-11eb-91a2-670d995e8181.png)

**Version Info**

NuGet package version: 
[Microsoft.UI.Xaml 2.4.0]
[Microsoft.WinUI 3.0.0-preview3.201113.0]

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| YES | YES |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| May 2020 Update (19041)            | YES   |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | YES |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**
@gabbybilka asked me to make this issue, and this is the comment she mentioned... #913 

>Thank you for the note and the color suggestion! The toned down dark theme colors were motivated to keep consistency with the Fluent colors and a general more muted Windows palette. Digging a little deeper the contrast between the Warning and Success colors is only 1.05:1 which wasn't as obvious to me initially. Could you open an issue on the lack of contrast between the two Severity states? We can continue to iterate and find a solution there. Thank you!
>

**Comment by gabbybilka** [comment]
@YuliKl as FYI, the highest contrast between the four dark theme colors is 1.252:1 between the informational and success colors. Although we also have the icon to also indicate Severity, the colors are a lot lower in contrast than I realized initially.

**Comment by mdtauk** [comment]
Github's new Dark Theme has a green looking green colour.

![image](https://user-images.githubusercontent.com/7389110/101632561-fc857780-3a1d-11eb-9016-bdb5a3e50af8.png)

**Comment by tipa** [comment]
I'm also not happy with the current InfoBar colors. The light theme backgrounds are too light and the dark theme backgrounds are too dark. That's without having measured any hues or contrasts (and I assume the users of my app aren't doing that either), the colors just "feel" weird.
If there's still a chance to redesign the InfoBar then I would also support the suggestion @mdtauk make here: https://github.com/microsoft/microsoft-ui-xaml/issues/913#issuecomment-506340446 - with the severity color not being the background behind the text the color can actually "express" its meaning and doesn't have to be lighter or darker just for readability's sake. And GitHub seems to use the same approach with their "info bars".
If redesigning the InfoBar is not an option, I would still appreciate darker colors in Light Theme and brighter colors in Dark Theme

**Comment by mdtauk** [comment]
@tipa   The design chosen matches the Fluent Web control which predates this one.

I think the GitHub info banner is a good middle ground between the mockups I did, and the current designs.  It would also help with contrast ratios, as the text is on a black/very dark or white/very light background, and the icon area features the more subtle background colour, with the foreground colour being prominent.

https://developer.microsoft.com/en-us/fluentui#/controls/web/messagebar

**Comment by tipa** [comment]
Some more examples of better colors, taken from https://docs.microsoft.com/

### dark theme
css class `is-danger`
![image](https://user-images.githubusercontent.com/2863198/101770277-89950300-3ae8-11eb-985a-92401114fea8.png)
css class `is-warning`
![image](https://user-images.githubusercontent.com/2863198/101770294-90237a80-3ae8-11eb-994f-8d85a89597ff.png)
css class `is-info`
![image](https://user-images.githubusercontent.com/2863198/101770338-9d406980-3ae8-11eb-8a9f-36212a07b115.png)
css class `is-success`
![image](https://user-images.githubusercontent.com/2863198/101770398-b0533980-3ae8-11eb-99e8-97788c85f5c9.png)

### light theme
css class `is-danger`
![image](https://user-images.githubusercontent.com/2863198/101770858-63bc2e00-3ae9-11eb-8e22-b582d11f2151.png)
css class `is-warning`
![image](https://user-images.githubusercontent.com/2863198/101770845-5d2db680-3ae9-11eb-9373-c54ed44f0af9.png)
css class `is-info`
![image](https://user-images.githubusercontent.com/2863198/101770866-69197880-3ae9-11eb-8390-38f631957121.png)
css class `is-success`
![image](https://user-images.githubusercontent.com/2863198/101770876-6f0f5980-3ae9-11eb-8628-aa1973227b9b.png)

**Comment by YuliKl** [comment]
Thank you for the suggestions. 

@gabbybilka, there's no color contrast ratio requirement that comes into play here (as long as the text remains legible and at 4.5:1 against its background, which is the case).

**Comment by mdtauk** [comment]
You may want to reconsider the InfoBar colour used by default, now that the Page Background colour has changed for the Light Theme. **(This is when using the Latest visuals)**

![image](https://user-images.githubusercontent.com/7389110/104039194-f70e7f00-51cd-11eb-836b-e201786042fd.png)

The Dark Theme is also hard to distinguish too

![image](https://user-images.githubusercontent.com/7389110/104039297-1c02f200-51ce-11eb-84d6-66ecf75f678b.png)

@YuliKl @gabbybilka @chigy
